### PR TITLE
Drop 'Enabling cpu compatibility enforcement'

### PR DESCRIPTION
**Pull request by vasiliy-ul**
**What this PR does / why we need it**:

The corresponding feature gate `CPUNodeDiscovery` is now enabled by default, hence the section is not needed anymore.

https://github.com/kubevirt/kubevirt/pull/7728

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/virtual_machines/virtual_hardware.md [hunk 1]
```diff
@@ -136,26 +136,18 @@ spec:
...
-
-This feature-gate allows kubevirt to take VM cpu model and cpu features
-and create node selectors from them. With these node selectors, VM can
-be scheduled on the node which can support VM cpu model and features.
-
 #### Labeling nodes with cpu models and cpu features
 
+KubeVirt takes VM cpu model and features to create node selectors out of them. With
+the node selectors, VM can be scheduled on the nodes which can support VM cpu model
+and features.
```
Comment (aburdenthehand) [inline docs/virtual_machines/virtual_hardware.md:143]:
```suggestion
KubeVirt can create node selectors based on VM cpu models and features. With
these node selectors, VMs will be scheduled on the nodes that support the matching VM cpu model
and features.
```

**Review by aburdenthehand — COMMENTED** [review]
Nice one. 
It took me a few reads to get it so I've suggested a couple of changes to help people like me who are unfamiliar with some of this.

File: docs/virtual_machines/virtual_hardware.md (hunk 1 above)
Comment (vasiliy-ul) [reply docs/virtual_machines/virtual_hardware.md]: Done. Now it is much better :+1:

**Comment by aburdenthehand** [comment]
/lgtm
Thanks @vasiliy-ul

**Review by phoracek — APPROVED** [review]
/approve

**Pull request merged**